protected/humhub/modules/space/widgets/MembershipButton.php

```diff
@@ -20,20 +23,78 @@ class MembershipButton extends Widget
 {
 
     /**
-     * @var \humhub\modules\space\models\Space
+     * @var Space
      */
     public $space;
 
+    /**
+     * @var array Options buttons
+     */
+    public $options = [];
+
+    private function getDefaultOptions()
+    {
+        return [
+            'requestMembership' => [
+                'title' => Yii::t('SpaceModule.base', 'Join'),
+                'attrs' => ['id' => 'requestMembershipButton', 'class' => 'btn btn-info', 'data-target' => '#globalModal'],
+            ],
+            'becomeMember' => [
+                'title' => Yii::t('SpaceModule.base', 'Join'),
+                'attrs' => ['id' => 'requestMembershipButton', 'class' => 'btn btn-info', 'data-method' => 'POST'],
+            ],
+            'acceptInvite' => [
+                'title' => Yii::t('SpaceModule.base', 'Accept Invite'),
+                'attrs' => ['class' => 'btn btn-info', 'data-method' => 'POST'],
+                'groupClass' => 'btn-group',
+                'togglerClass' => 'btn btn-info',
+            ],
+            'declineInvite' => [
+                'title' => Yii::t('SpaceModule.base', 'Decline Invite'),
+                'attrs' => ['data-method' => 'POST'],
+            ],
+            'cancelPendingMembership' => [
+                'title' => '<span class="glyphicon glyphicon-time"></span>&nbsp;&nbsp;' . Yii::t('SpaceModule.base', 'Pending'),
+                'attrs' => [
+                    'data-method' => 'POST',
+                    'data-confirm' => Yii::t('SpaceModule.base', 'Would you like to withdraw your request to join Space {spaceName}?', ['{spaceName}' => '"' . $this->space->getDisplayName() . '"']),
+                    'class' => 'btn btn-info active',
+                ],
+            ],
+            'member' => [
+                'title' => '<span class="glyphicon glyphicon-ok"></span>&nbsp;&nbsp;' . Yii::t('SpaceModule.base', 'Member'),
+                'attrs' => [
+                    'data-method' => 'POST',
+                    'data-confirm' => Yii::t('SpaceModule.base', 'Would you like to end your membership in Space {spaceName}?', ['{spaceName}' => '"' . $this->space->getDisplayName() . '"']),
```

Comment:
Html::encode?

Reply:
answer in https://github.com/humhub/humhub/pull/5057#discussion_r644786497